src/spetlr/etl/loaders/simple_loader.py

```diff
@@ -15,20 +15,29 @@ def append(self, df: DataFrame) -> None:
         pass
 
 
+class Upsertable(Protocol):
+    def upsert(self, df: DataFrame, join_cols: List[str]) -> Union[DataFrame, None]:
+        pass
+
+
 class SimpleLoader(Loader):
     def __init__(
         self,
-        handle: Union[Overwritable, Appendable],
+        handle: Union[Overwritable, Appendable, Upsertable],
         *,
         mode: str = "overwrite",
+        join_cols: List[str] = None,
         dataset_input_keys: Union[str, List[str]] = None,
```

Comment:
Maybe now that we are moving away from the "both string and list" type for the dataset input keys, we could update the type hint. Mind, updating a hint does not break any of the subsequent flows if they provide another type as long as the underlying logic (read: `super().init(...)`) handles both types.

Suggestion: change type hint from `Union[str, List[str]]` to `List[str]]`

Reply:
Agree. Implemented.